### PR TITLE
pkg/nimble/netif: use global MSYS memory pool

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
In my initial implementation of `nimble_netif` I decided for using a local memory pool for buffering IP packets when transferring them from GNRC `pktbuf` to NimBLEs own (`mbuf`) memory pool. Now after more understanding and a lot of more experience using NimBLE, I come to the conclusion that it is far sufficient to use NimBLE global memory pool for this step as well. 

This saves quite a bit of otherwise over provisioned memory. Example - building `gnrc_networking` for `nrf52dk`:
without this PR:
```
   text    data     bss     dec 
 155788     288   46620  202696
```
with this PR:
```
   text    data     bss     dec 
 155796     288   38180  194264  
```
--> So for the default configuration (max connections := 3, mbuf-pool size := 35) **we save >8kb RAM!**

To further justify this change, the following diagram shows NimBLEs maximum buffer usage per node over time in an IP-over-BLE multi-hop experiment with 15 nodes:
em6_putnon_statconn-static_1s1h39b_i2000_20201201-191422_buf_msys
This chart was taken with the memory-wise most demanding parameters that I used so far and it can clearly be seen that no Node ever used more than ~27 of its available buffers in the pool.

If ever running into trouble of exceeding buffer space, this change makes it also much more simple to increase the overvall memory size, as one has only to change a single value instead of tuning MSYS and the local memory pool individually...

Bottom line: we should be save with removing the local buffers in `nimble_netif` in favor of using NimBLEs global `MSYS` pool instead.

### Testing procedure
I recommend running the `gnrc_networking` example with 2 or more `nrf52dk` nodes and verify that IP communication between the nodes still works as expected (remember to use the `ble` shell command for connecting nodes...). This can nicely be done in the `iotlab`.

Alternatively one could verify this change also by running the `tests/nimble_l2cap` test using two supported nodes.

### Issues/PRs references
nonde

### CODE REVIEW

File: pkg/nimble/netif/nimble_netif.c [hunk 1]
```diff
@@ -442,16 +429,22 @@ static int _on_gap_master_evt(struct ble_gap_event *event, void *arg)
...
             _on_gap_connected(conn, event->connect.conn_handle);
             conn->state |= NIMBLE_NETIF_GAP_MASTER;
 
-            struct os_mbuf *sdu_rx = os_mbuf_get_pkthdr(&_mbuf_pool, 0);
-            /* we should never run out of buffer space... */
-            assert(sdu_rx != NULL);
+            struct os_mbuf *sdu_rx = os_msys_get_pkthdr(
+                                            MYNEWT_VAL_BLE_L2CAP_COC_MPS, 0);
+            /* in the rare case that we run out of buffers, we close the new
+             * connection right right away */
```
Comment (janosbrodbeck) [inline pkg/nimble/netif/nimble_netif.c:439]:
```suggestion
             * connection right away */
```
very minor but stumbled upon it while looking through the changes

**Comment by haukepetersen** [comment]
rebased and fixed typo

**Comment by haukepetersen** [comment]
rebased.

Anyone available to ACK this change?

File: pkg/nimble/netif/nimble_netif.c [hunk 2]
```diff
@@ -442,16 +429,22 @@ static int _on_gap_master_evt(struct ble_gap_event *event, void *arg)
...
-            /* we should never run out of buffer space... */
-            assert(sdu_rx != NULL);
+            struct os_mbuf *sdu_rx = os_msys_get_pkthdr(
+                                            MYNEWT_VAL_BLE_L2CAP_COC_MPS, 0);
+            /* in the rare case that we run out of buffers, we close the new
+             * connection right away */
+            if (sdu_rx == NULL) {
+                ble_gap_terminate(handle, BLE_ERR_REM_USER_CONN_TERM);
+                break;
+            }
```
Comment (fjmolinas) [inline pkg/nimble/netif/nimble_netif.c:443]: Why did this change from "we should never run out of buffer space..." to "in the rare case that we run out of buffers", just future proofing?

File: pkg/nimble/netif/nimble_netif.c [hunk 3]
```diff
@@ -442,16 +429,22 @@ static int _on_gap_master_evt(struct ble_gap_event *event, void *arg)
                 _notify(handle, NIMBLE_NETIF_ABORT_MASTER, addr);
                 return 0;
             }
+
```
Comment (fjmolinas) [inline pkg/nimble/netif/nimble_netif.c:432]: unrelated?

**Review by fjmolinas — COMMENTED** [review]
> In my initial implementation of nimble_netif I decided for using a local memory pool for buffering IP packets when transferring them from GNRC pktbuf to NimBLEs own (mbuf) memory pool. Now after more understanding and a lot of more experience using NimBLE, I come to the conclusion that it is far sufficient to use NimBLE global memory pool for this step as well.|

Can you explain why initially you though buffering was needed? Otherwise looks good.

File: pkg/nimble/netif/nimble_netif.c (hunk 2 above)
Comment (haukepetersen) [reply pkg/nimble/netif/nimble_netif.c:443]: because before we used buffer space that was solely allocated and sized for this specific purpose while now we use shared buffers that could potentially be exhausted by other modules

**Comment by haukepetersen** [comment]
rebased and reverted newline

File: pkg/nimble/netif/nimble_netif.c (hunk 3 above)
Comment (haukepetersen) [reply pkg/nimble/netif/nimble_netif.c]: reverted

**Comment by fjmolinas** [comment]
Tested everything still works:

```
2021-04-12 15:18:33,235 # ble scan
2021-04-12 15:18:33,238 # scanning (for 500ms) ...
2021-04-12 15:18:33,739 # done
2021-04-12 15:18:33,747 # [ 0] 41:e0:4d:16:9e:2e (RANDOM) [IND] "undefined", adv_msg_cnt: 2, adv_int: 307us, last_rssi: -79
2021-04-12 15:18:33,755 # [ 1] 88:c6:26:c4:70:eb (PUBLIC) [IND] "", adv_msg_cnt: 12, adv_int: 35605us, last_rssi: -69
2021-04-12 15:18:33,765 # [ 2] f8:da:0c:58:5e:1c (PUBLIC) [SCAN_IND] "undefined", adv_msg_cnt: 11, adv_int: 24006us, last_rssi: -71
2021-04-12 15:18:33,773 # [ 3] 61:40:bd:86:c2:09 (RANDOM) [NONCONN_IND] "undefined", adv_msg_cnt: 3, adv_int: 106084us, last_rssi: -81
2021-04-12 15:18:33,783 # [ 4] 4c:d9:d1:e8:67:78 (RANDOM) [NONCONN_IND] "undefined", adv_msg_cnt: 3, adv_int: 108534us, last_rssi: -85
2021-04-12 15:18:33,791 # [ 5] 7c:7c:93:7c:6f:ec (RANDOM) [IND] "undefined", adv_msg_cnt: 2, adv_int: 305us, last_rssi: -88
> ble scan
2021-04-12 15:18:38,539 # ble scan
2021-04-12 15:18:38,541 # scanning (for 500ms) ...
2021-04-12 15:18:39,042 # done
2021-04-12 15:18:39,051 # [ 0] 74:32:55:a6:3b:7e (RANDOM) [IND] "undefined", adv_msg_cnt: 4, adv_int: 70155us, last_rssi: -82
2021-04-12 15:18:39,059 # [ 1] 88:c6:26:c4:70:eb (PUBLIC) [IND] "", adv_msg_cnt: 12, adv_int: 34195us, last_rssi: -69
2021-04-12 15:18:39,067 # [ 2] c2:13:f0:9b:61:f6 (RANDOM) [IND] "bleRIOT", adv_msg_cnt: 3, adv_int: 158864us, last_rssi: -26
2021-04-12 15:18:39,076 # [ 3] 7c:7c:93:7c:6f:ec (RANDOM) [IND] "undefined", adv_msg_cnt: 1, adv_int: 0us, last_rssi: -86
2021-04-12 15:18:39,084 # [ 4] 4c:d9:d1:e8:67:78 (RANDOM) [NONCONN_IND] "undefined", adv_msg_cnt: 1, adv_int: 0us, last_rssi: -84
2021-04-12 15:18:39,094 # [ 5] 7f:1b:68:e2:d7:6b (RANDOM) [NONCONN_IND] "undefined", adv_msg_cnt: 2, adv_int: 54062us, last_rssi: -89
2021-04-12 15:18:39,104 # [ 6] f8:da:0c:58:5e:1c (PUBLIC) [SCAN_IND] "undefined", adv_msg_cnt: 12, adv_int: 30962us, last_rssi: -70
2021-04-12 15:18:39,113 # [ 7] 61:40:bd:86:c2:09 (RANDOM) [NONCONN_IND] "undefined", adv_msg_cnt: 4, adv_int: 78115us, last_rssi: -63
2021-04-12 15:18:39,122 # [ 8] 41:e0:4d:16:9e:2e (RANDOM) [IND] "undefined", adv_msg_cnt: 4, adv_int: 68591us, last_rssi: -63
> ble connect c2:13:f0:9b:61:f6
2021-04-12 15:18:45,507 # ble connect c2:13:f0:9b:61:f6
2021-04-12 15:18:45,512 # initiated connection procedure with C2:13:F0:9B:61:F6
> 2021-04-12 15:18:45,926 # event: handle 0 -> CONNECTED as MASTER (C2:13:F0:9B:61:F6)
ping6 fe80::c213:f0ff:fe9b:61f6
2021-04-12 15:18:52,613 # ping6 fe80::c213:f0ff:fe9b:61f6
2021-04-12 15:18:52,678 # 12 bytes from fe80::c213:f0ff:fe9b:61f6%8: icmp_seq=0 ttl=64 time=59.220 ms
2021-04-12 15:18:53,678 # 12 bytes from fe80::c213:f0ff:fe9b:61f6%8: icmp_seq=1 ttl=64 time=74.696 ms
2021-04-12 15:18:54,678 # 12 bytes from fe80::c213:f0ff:fe9b:61f6%8: icmp_seq=2 ttl=64 time=90.059 ms
2021-04-12 15:18:54,679 #
2021-04-12 15:18:54,683 # --- fe80::c213:f0ff:fe9b:61f6 PING statistics ---
2021-04-12 15:18:54,688 # 3 packets transmitted, 3 packets received, 0% packet loss
2021-04-12 15:18:54,692 # round-trip min/avg/max = 59.220/74.658/90.059 ms
```

**Review by fjmolinas — APPROVED** [review]
ACK, please squash

**Comment by haukepetersen** [comment]
squashed

**Comment by haukepetersen** [comment]
All green -> go.
